.github/workflows/backend_cd.yml

```diff
@@ -6,12 +6,27 @@ on:
 env:
   env_file: ${{ secrets.env_prod }}
   aws_pem_file: ${{ secrets.aws_pem_file }}
-  docker_password: ${{ secrets.docker_password }}
 jobs:
   Testing:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           ref: main 
-      - run: ls && echo "$env_file" > ".env" && echo "$aws_pem_file" > aws.pem && docker login --username swanso15 --password $docker_password && docker build -t swanso15/cupboard_backend:latest . && docker image ls && docker save -o cupboard_backend.tar swanso15/cupboard_backend:latest
+      - name: Create docker image and send to aws
+        run: |
+          echo "$env_file" > ".env"
+          echo "$aws_pem_file" > aws.pem
+          chmod 600 aws.pem
+          docker build -t cupboard_backend:latest .
+          docker save -o cupboard_backend.tar cupboard_backend:latest
+          scp -i aws.pem -o StrictHostKeyChecking=no cupboard_backend.tar ec2-user@35.183.197.0:/home/ec2-user
+      - name: Load and run docker image on aws
+        run: |
+          ssh -i aws.pem -o StrictHostKeyChecking=no ec2-user@35.183.197.0 "
+            docker ps -a -q | xargs -r docker stop
+            docker ps -a -q | xargs -r docker remove
+            docker image rm --force cupboard_backend
+            docker image load --input cupboard_backend.tar
+            docker run -d -p 6060:6060 cupboard_backend:latest
```

Comment:
Right now, the docker image is being built with the .env file. This is not a good practice because this means we need to rebuild the entire docker image every time an environment variable changes i.e. password changes and this can end up costing money due to computation time.

Instead, do the following:
1. Add a `.dockerignore` file (similar to `.gitignore`) with `.env` and `.git` in the file.
2. Have the `.env` file in the EC2 container. You can just scp the environment variable file.
3. Change the `docker run -d -p 6060:6060 cupboard_backend:latest` to `docker run -d -p 6060:6060 --env-file .env cupboard_backend:latest`
This command is saying run the docker image on port 6060 with the env file.

This way, if we ever have to just change the .env file, then you can do the run command with the updated .env file rather than rebuilding the whole image.